### DISCUSSION TITLE
Embedding the unmanaged libraries

### DISCUSSION THREAD

**Discussion by RobSwDev** [desjarlais/Scintilla.NET#42] [Q&A] [answered]
It seems embedding the unmanged libraries was a deliberate choice, described in https://github.com/VPKSoft/Scintilla.NET#one-library
I disagree with some of the reasoning however. It's perfectly possible to create a nuget package that deploys unmanged libraries to the correct locations. Possibly things have changed since the choice was originally made

Automatically extracting these to the temp directory causes problems for us in some ultra-restricted environments.
Previously we've been manually extracting the binaries and creating our own nuget package (using Scintilla.SetModulePath), but this makes it much harder to upgrade Scintilla.

**Comment by VPKSoft** [comment, marked as answer]
Hi,
You are right to disagree - this design choice and maintaining it is due to the history of the project.
1. The project at CodePlex
2. The original GitHub implementation (abandoned): https://github.com/jacobslusser/ScintillaNET
3. This one

I'll probably remove the ZIP-phase in the near future... I also accept PR's 🙂

**Reply by RobSwDev** [reply]
Hi Petteri,
I've created a PR to remove the embedded dlls. However, I branched at commit 73fd0cfdebc8518cd0fff33e59138ac453d41bef (tag: v.5.3.2.1)
so there are conflicts against the head of master.

What did you mean by 
"Add abstraction layer for possible multi-platform implementation"
E.g. does platform here potentially include Linux/Android/etc?

**Comment by VPKSoft** [comment]
Hi,
Thanks for the PR!
I rebranched the master to the v.5.3.2.1 and moved the abstraction changes into develop branch. If you would change the target branch of the pull request?
The idea is to create a Linux and macOS and a separate Windows version to use an abstraction NuGet and create a cross platform implementation for the [Eto](https://github.com/picoe/Eto) framework into a different repository: https://github.com/VPKSoft/Scintilla.NET.Eto. The WinForms implementation would also use the same abstraction NuGet in the future versions when/if the work is done.

**Comment by desjarlais** [comment]
This seems to be answered, so I'm not sure what more of a discussion is needed.  Feel free to reopen or create another discussion post if there is more work in this area that is needed.
